crates/provider/src/builder.rs

```diff
@@ -437,11 +439,17 @@ impl<L, F> ProviderBuilder<L, F, Ethereum> {
         let anvil_layer = crate::layers::AnvilLayer::from(f(Default::default()));
         let url = anvil_layer.endpoint_url();
 
-        let wallet = alloy_signer_wallet::Wallet::from(anvil_layer.instance().keys()[0].clone());
+        let default_keys = anvil_layer.instance().keys().to_vec();
+        let (default_key, remaining_keys) = default_keys.split_first().expect("no keys available");
```

Comment:
prevent panic from having 0 keys when calling `on_anvil_with_wallet_and_config`

This can panic if the `f` function sets accounts to 0. This is not concerning (and is the pre-existing behavior) but we should probably fix it

followup work (not this PR)

Reply:
https://github.com/alloy-rs/alloy/issues/889